cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -131,22 +132,7 @@ func (k *proxy) ValidateKubernetesVersion() error {
 
 // GetConfig returns the config for a kubernetes client.
 func (k *proxy) GetConfig() (*rest.Config, error) {
-	config, err := k.configLoadingRules.Load()
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to load Kubeconfig")
-	}
-
-	configOverrides := &clientcmd.ConfigOverrides{
-		CurrentContext: k.kubeconfig.Context,
-		Timeout:        k.timeout.String(),
-	}
-	restConfig, err := clientcmd.NewDefaultClientConfig(*config, configOverrides).ClientConfig()
-	if err != nil {
-		if strings.HasPrefix(err.Error(), "invalid configuration:") {
-			return nil, errors.New(strings.Replace(err.Error(), "invalid configuration:", "invalid kubeconfig file; clusterctl requires a valid kubeconfig file to connect to the management cluster:", 1))
-		}
-		return nil, err
-	}
+	restConfig := ctrl.GetConfigOrDie()
```

Comment:
I think the issue with this is, the underlying function tries to use a global kubeconfig flag, which we don't set or use.

Instead I think what we should do is use `rest.InCLusterConfig()` in the error case as fallback.

Reply:
Updated. Anyway, though it would take a lot of refactoring, wouldn't it be better to use kubeconfig flag?